test/gtest/fold.hpp

```diff
@@ -0,0 +1,303 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "../driver/tensor_driver.hpp"
+#include "cpu_unfold.hpp"
+#include "get_handle.hpp"
+#include "miopen/allocator.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/fold.hpp>
+
+struct FoldTestCase
+{
+    size_t N;
+    size_t C;
+    size_t D;
+    size_t H;
+    size_t W;
+    std::vector<int32_t> outputSize;
+    std::vector<int32_t> kernelSize;
+    std::vector<int32_t> stride;
+    std::vector<int32_t> padding;
+    std::vector<int32_t> dilation;
+    bool isContiguous = true;
+    friend std::ostream& operator<<(std::ostream& os, const FoldTestCase& tc)
+    {
+        os << "N:" << tc.N << " C:" << tc.C << " D:" << tc.D << " H:" << tc.H << " W:" << tc.W;
+        os << " output_size:";
+        for(const auto& outs : tc.outputSize)
+            os << outs << " ";
+        os << " kernel_size:";
+        for(const auto& ks : tc.kernelSize)
+            os << ks << " ";
+        os << "stride:";
+        for(const auto& s : tc.stride)
+            os << s << " ";
+        os << "padding:";
+        for(const auto& p : tc.padding)
+            os << p << " ";
+        os << "dilation:";
+        for(const auto& d : tc.dilation)
+            os << d << " ";
+        os << "isContiguous:" << std::boolalpha << tc.isContiguous;
+        return os;
+    }
+
+    std::vector<size_t> GetInput()
+    {
+        if((N != 0) && (C != 0) && (D != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, D, H, W});
+        }
+        else if((N != 0) && (C != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, H, W});
+        }
+        else if((N != 0) && (C != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, W});
+        }
+        else if((N != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, W});
+        }
+        else if((N != 0))
+        {
+            return std::vector<size_t>({N});
+        }
+        else
+        {
+            std::cout << "Error Input Tensor Lengths\n" << std::endl;
+            return std::vector<size_t>({0});
+        }
+    }
+
+    std::vector<size_t> ComputeStrides(std::vector<size_t> inputDim) const
+    {
+        if(!isContiguous)
+            std::swap(inputDim.front(), inputDim.back());
+        std::vector<size_t> strides(inputDim.size());
+        strides.back() = 1;
+        for(int i = inputDim.size() - 2; i >= 0; --i)
+            strides[i] = strides[i + 1] * inputDim[i + 1];
+        if(!isContiguous)
+            std::swap(strides.front(), strides.back());
+        return strides;
+    }
+};
+
+std::vector<FoldTestCase> FoldTestConfigs()
+{ // n c d h w padding
+    return {
+        {3, 3 * 2 * 2, 0, 0, 3 * 4, {4, 5}, {2, 2}, {1, 1}, {0, 0}, {1, 1}, true},
```

Comment:
Please disable clang-format (with `// clang-format: off`) to prevent breakage later

Reply:
resolve in commit [27e26c3](https://github.com/moreh-dev/MIOpen/pull/37/commits/27e26c35b5d2fedee5bee62202e300f4d2665d40)